code/game/objects/effects/alien_acid.dm

```diff
@@ -15,73 +15,89 @@
 	density = 0
 	opacity = 0
 	anchored = 1
-	unacidable = 1
-	var/atom/target
-	var/ticks = 0
-	var/target_strength = 0
+	acid_state = UNACIDABLE
+	var/acid_type = "alienacid"
+	var/turf/target
+	var/target_strength = 60 //phil235 roughly 1 minute for alien acid on wall, if 1 process per second
 
 
-/obj/effect/acid/New(loc, targ)
+/obj/effect/acid/New(loc, acid_pwr, acid_amt, acid_id)
 	..(loc)
-	target = targ
-
-	//handle APCs and newscasters and stuff nicely
-	pixel_x = target.pixel_x
-	pixel_y = target.pixel_y
-
-	if(isturf(target))	//Turfs take twice as long to take down.
-		target_strength = 640
-	else
-		target_strength = 320
-	tick()
 
+	target = get_turf(src)
 
-/obj/effect/acid/proc/tick()
-	if(!target)
-		qdel(src)
+	if(acid_id)
+		acid_type = acid_id
 
-	ticks++
+	if(acid_amt)
+		acid_level = min(acid_amt*acid_pwr, 12000)
```

Comment:
Can you leave a comment explaining what this number represents, and why you capped it?